aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -1321,4 +1322,157 @@ Tensor any_sparse(const Tensor& self) {
+
+  TORCH_CHECK(self.size(0) == mat2.size(0), "bmm_sparse: 'self.size(0)' and 'mat2.size(0)' must match");
+  TORCH_CHECK(self.size(2) == mat2.size(1), "bmm_sparse: 'self.size(2)' and 'mat2.size(1)' must match");
+
+  result.resize_({self.size(0), self.size(1), mat2.size(2)});
+
+  if (self._nnz() == 0) {
+    result.zero_();
+    return result;
+  }
+
+  // First need to coalesce to get all of the first dimension indices
+  // in order since we'll be sending each matrix into the MM operation
+  SparseTensor self_coalesced;
+
+  if (!self.is_coalesced()){
+    self_coalesced = self.coalesce();
+  } else {
+    self_coalesced = self;
+  }
+
+  int64_t nnz =        self_coalesced._nnz();
+  LongTensor indices = self_coalesced._indices();
+  Tensor values =      self_coalesced._values();
+
+  LongTensor indices_dim0 = indices[0];
+  int64_t indices_dim0_stride = indices_dim0.stride(0);
+  int64_t* indices_dim0_ptr = indices_dim0.data_ptr<int64_t>();
+  LongTensor indices_dim1_dim2 = indices.slice(0, 1, 3);
+
+  int64_t dim_i = self_coalesced.size(1);
+  int64_t dim_j = self_coalesced.size(2);
+  int64_t dim_k = mat2.size(2);
+
+  Scalar beta = 0;
+  Tensor t = at::ones({self_coalesced.size(1)});
+  Scalar alpha = 1;
+
+  int64_t mat_el_begin_idx = 0;
+
+  int64_t num_matrices = self_coalesced.size(0);
+
+  // Iterate through each set of 2D matrices within the 3D
+  // tensor inputs, performing a matrix multiply with each one.
+  int64_t start_mat_num = indices_dim0_ptr[0];
+  for (int64_t cur_mat_num = 0;
+    (cur_mat_num < num_matrices);
+    cur_mat_num++
+  ) {
+    // If there are sparse matrices at the beginning or end that
+    // have all zero elements, we need to zero out the result matrix.
+    if ((cur_mat_num < start_mat_num) || (mat_el_begin_idx >= nnz)) {
+      result[cur_mat_num].zero_();
+      continue;
+    }
+
+    // Search for the range of sparse tensor elements that
+    // correspond to the current matrix number. We already know
+    // where the current matrix begins, so we just need to find
+    // the end. The search excludes everything to the left of
+    // the starting point, for best performance
+    bool mat_end_found;
+    int64_t mat_el_end_idx = binary_search_strided_rightmost(
+      cur_mat_num,
+      indices_dim0_ptr+(mat_el_begin_idx*indices_dim0_stride),
+      indices_dim0_stride,
+      nnz-mat_el_begin_idx,
+      &mat_end_found
+    ) + mat_el_begin_idx;
+
+    if (mat_end_found) {
+      mat_el_end_idx++;
+
+      // Create tensors to view just the current set of matrices
+      const Tensor dense_matrix = mat2[cur_mat_num];
+      Tensor result_matrix = result[cur_mat_num];
+      LongTensor sparse_indices = indices_dim1_dim2.slice(1, mat_el_begin_idx, mat_el_end_idx);
+      Tensor sparse_values = values.slice(0, mat_el_begin_idx, mat_el_end_idx);
+      int64_t sparse_nnz = mat_el_end_idx - mat_el_begin_idx;
+
+      AT_DISPATCH_ALL_TYPES(
```

Comment:
Similarly, prefer not to repeatedly dispatch inside of the batch loop.

Reply:
Done